Desktop/Prework/Allegro/allegro/src/App.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import ListItem from './components/ListItem/ListItem';
+import Pagination from './components/Pagination/Pagination';
+import styles from './App.module.scss';
+import Header from './components/Header/Header';
+import Loader from './components/Loader/Loader'
+
+
+
+const Pokedex = require('pokedex-promise-v2');
+
+const options = [
+  { value: 'normal', label: 'normal' },
+  { value: 'fighting', label: 'fighting' },
+  { value: 'flying', label: 'flying' },
+  { value: 'poison', label: 'poison' },
+  { value: 'ground', label: 'ground' },
+  { value: 'rock', label: 'rock' },
+  { value: 'bug', label: 'bug' },
+  { value: 'ghost', label: 'ghost' },
+  { value: 'steel', label: 'steel' },
+  { value: 'fire', label: 'fire' },
+  { value: 'water', label: 'water' },
+  { value: 'grass', label: 'grass' },
+  { value: 'electric', label: 'electric' },
+  { value: 'psychic', label: 'psychic' },
+  { value: 'ice', label: 'ice' },
+  { value: 'dragon', label: 'dragon' },
+  { value: 'dark', label: 'dark' },
+  { value: 'fairy', label: 'fairy' },
+  { value: 'unknown', label: 'unknown' },
+  { value: 'shadow', label: 'shadow' }
+]
+
+class App extends React.Component {
+
+  state = {
+    offset: 0,
+    pokemonInfos: [],
+    filtredPokemonInfos: [],
+    perPage: 8,
+    currentPage: 0,
+    pageCount: null,
+    selectedOption: null,
+    loading: true
+  };
+
+  componentDidMount() {
+    this.pokemonGetter()
+  }
+
+  pokemonGetter = () => {
+    const P = new Pokedex();
+
+    P.getPokemonsList()
+      .then(res => {
+        return res
+      }).then(res => {
+        res.results.map(pokemon => {
+          return pokemon.name
+        }).map(pokemonName => {
+          return P.getPokemonByName(pokemonName)
+            .then(res => {
+              const pokemonsInfos = [...this.state.pokemonInfos];
+              this.setState({
+                pokemonInfos: [
+                  ...pokemonsInfos, res],
+                filtredPokemonInfos: [
+                  ...pokemonsInfos, res],
+                loading: false
+              });
+            })
+        })
+      }).catch(err => {
+        console.log(err)
+      });
+
+    this.updatePageCount()
```

Comment:
Promises run asynchronously, but this method is used in a synchronous manner. It means that it might use `pokemonInfos` before it is resolved in the promise above. async/await should resolve this issue